### PR TITLE
Fixes #8261: Warn when the string length reaches the maximum value supported by the agent

### PR DESCRIPTION
**Pull request by RaphaelGauthier**
https://www.rudder-project.org/redmine/issues/8261

### CODE REVIEW

File: builder/index.html [hunk 1]
```diff
@@ -62,11 +62,14 @@ <h3 class="modal-title">{{actionName}} {{kind}}</h3>
               <h3 class="modal-title">Clone Technique</h3>
             </div>
             <div class="modal-body">
-              <div class="form-group">
+              <div class="form-group" ng-class="{'has-error' : !checkLength(technique.name)}">
                 <label for="cloneName" class="control-label">New Technique name:</label>
                 <div>
                   <input  type="text" id="cloneName" name="name" techniquename class="form-control" ng-model="technique.name" placeholder="Technique Name" required>
```
Comment (VinceMacBuche) [inline builder/index.html]: Here you could have put ng-maxlenght = 2048 character

Comment (VinceMacBuche) [inline builder/index.html]: https://docs.angularjs.org/api/ng/directive/input

File: builder/index.html [hunk 2]
```diff
@@ -62,11 +62,14 @@ <h3 class="modal-title">{{actionName}} {{kind}}</h3>
...
                 </div>
+                <span class="text-danger" ng-show="!checkLength(technique.name)">
```
Comment (VinceMacBuche) [inline builder/index.html]: And using ng-maxlength, you can use here :  ng-show="formName.inputName.$errors.maxlength"

File: builder/index.html [hunk 3]
```diff
@@ -164,18 +167,24 @@ <h3 >{{originalTechnique.name}} <span ng-hide="originalTechnique.name">New Techn
                                 <i class="glyphicon" ng-class="{'glyphicon-chevron-down': isopen, 'glyphicon-chevron-right': !isopen}"></i>
                                 General information
                             </accordion-heading>
-                            <div class="form-group" show-errors>
+                            <div class="form-group" show-errors ng-class="{'has-error' : !checkShortLength(selectedTechnique.name)}">
```
Comment (VinceMacBuche) [inline builder/index.html]: Same feedbad here :)

File: builder/index.html [hunk 4]
```diff
@@ -164,18 +167,24 @@ <h3 >{{originalTechnique.name}} <span ng-hide="originalTechnique.name">New Techn
...
                                 <label for="techniqueName" class="col-sm-3 control-label">Name:</label>
                                 <div class="col-sm-8">
                                     <input  type="text" id="techniqueName" name="name" class="form-control" techniquename ng-model="selectedTechnique.name" placeholder="Technique Name" required>
                                 </div>
+                                <span class="text-danger col-sm-8 col-sm-offset-3" ng-show="!checkShortLength(selectedTechnique.name)">
```
Comment (VinceMacBuche) [inline builder/index.html]: Same

File: builder/index.html [hunk 5]
```diff
@@ -164,18 +167,24 @@ <h3 >{{originalTechnique.name}} <span ng-hide="originalTechnique.name">New Techn
...
+                                    Fields over 255 characters are currently not supported. Please insert your content into a file, and copy it with a file_copy_* method, or use a template.
+                                </span>
                                 <span class="text-danger col-sm-8 col-sm-offset-3" ng-show="editForm.name.$error.techniqueName">Technique name must be unique</span>
                             </div>
-                            <div class="form-group">
+                            <div class="form-group" ng-class="{'has-error' : !checkLength(selectedTechnique.description)}">
```
Comment (VinceMacBuche) [inline builder/index.html]: Same

File: builder/js/ncf.js [hunk 6]
```diff
@@ -62,7 +62,9 @@ app.directive('showErrors', function() {
       el.toggleClass('has-error', formCtrl[inputName].$invalid);
       // only apply the has-error class after the user leaves the text box
       inputNgEl.bind('blur', function() {
-        el.toggleClass('has-error', formCtrl[inputName].$invalid);
+        if(scope.checkLength(inputEl.value)){
```
Comment (VinceMacBuche) [inline builder/js/ncf.js]: No need ofthat check anymore

File: builder/js/ncf.js [hunk 7]
```diff
@@ -744,6 +746,20 @@ app.controller('ncf-builder', function ($scope, $modal, $http, $log, $location,
     window.top.location.reload();
   }
 
+  $scope.checkLength = function(str) {
```
Comment (VinceMacBuche) [inline builder/js/ncf.js]: Not needed anymore  (and would have been better with a checkLength(str,size) !!)

**Comment by RaphaelGauthier** [comment]
PR updated

File: builder/index.html [hunk 8]
```diff
@@ -231,7 +240,7 @@ <h5 class="pull-left method-param" >
                         <span class="glyphicon glyphicon-plus-sign" style="margin-right:5px"></span>
                         Add methods
                     </div>
-                    <button ng-disabled="editForm.$invalid || checkSelectedTechnique()" class="btn btn-default pull-right" ng-click="saveTechnique()">Save</button>
+                    <button ng-disabled="!checkInputLengthValidity() || editForm.$invalid || checkSelectedTechnique()" class="btn btn-default pull-right" ng-click="saveTechnique()">Save</button>
```
Comment (VinceMacBuche) [inline builder/index.html]: You can remove here checkInputLengthValidity

File: builder/js/ncf.js [hunk 9]
```diff
@@ -744,6 +744,12 @@ app.controller('ncf-builder', function ($scope, $modal, $http, $log, $location,
     window.top.location.reload();
   }
 
+  $scope.checkInputLengthValidity = function() {
```
Comment (VinceMacBuche) [inline builder/js/ncf.js]: You can remove this

**Comment by RaphaelGauthier** [comment]
PR updated

**Pull request merged**